addons/mass_mailing_sms/controllers/main.py

```diff
@@ -82,13 +85,18 @@ def blacklist_number(self, mailing_id, trace_code, **post):
         else:
             unsubscribe_error = _('Invalid number %s', post.get('sms_number', ''))
 
+        if unsubscribe_error:
+            params = werkzeug.urls.url_encode(
+                dict(**post, unsubscribe_error=unsubscribe_error)
+            )
+            return request.redirect(f'/sms/{mailing_id}/{trace_code}?{params}')
```

Comment:
Both routes serve different purposes—one renders the input field for entering the phone number, and the other handles blacklisting a number. While rendering the same template on both routes might seem to work visually, it would technically mean that the input field is being rendered on the validation route, which is not ideal. In my opinion, since we already have a dedicated route for rendering the number input template, we should keep it exclusive to that route. What do you think?

Reply:
Then can't we redirect to the same route when clicking on the button and check If it's an error show it else redirect to the unsubscribe route. What do you think about this?

Reply:
It's possible, and you're right. However, to achieve this, we would need to change the form action in the blacklist template and move the entire logic from the /unsubscribe route to the /sms route.

That said, if we’re restructuring the business logic, I don't think we even need a separate /unsubscribe route. Its sole purpose is to display a successful unsubscription message, which we can handle inline, just like we do for errors. What do you think? :)

Reply:
That is a big one. So let's wait for other validator's suggestions. (Keep this conversation open)

Reply:
Hello @datonefaridze If im not wrong Did you mean we should move the lines to 'blacklist_page" ? :)

Reply:
Absolutely! We can improve the code flow by moving the error checking to the blacklist_page. If there are no errors, we can successfully redirect to blacklist_number. 👍